### DISCUSSION TITLE
Is it possible to add a timestamp for the most recent publish date of a page?

### DISCUSSION THREAD

**Discussion by arretx** [squidfunk/mkdocs-material#3889] [Q&A]
When I update pages, it would be great if I could have the original date published and the most recent revision date at the bottom of the document, or at the top...or anywhere.  Is that possible?

**Comment by squidfunk** [comment]
The [`git-revision-date-localized`](https://squidfunk.github.io/mkdocs-material/setup/adding-a-git-repository/#revision-date-localized) plugin might be what you're looking for.

**Reply by bigrokm** [reply]
I'm NOT using git. **Is there a way for me to show the created and updated dates?**
https://squidfunk.github.io/mkdocs-material does it.
![image](https://github.com/user-attachments/assets/657da1a1-9629-4782-87f4-00fe80dff1c7)

**Reply by kamilkrzyskow** [reply]
@bigrokm 
This part of the page is rendered with the [source-file](https://github.com/squidfunk/mkdocs-material/blob/master/material/templates/partials/source-file.html) template, you could use [customization](https://squidfunk.github.io/mkdocs-material/customization/#extending-the-theme) and replace the logic in the template to read different values which you'd provide.

Also the data retrieval is handled over specific meta key-value pairs provided by the git-revision plugin, but you could also just input the information under the same key names yourself, so you don't have to customize / replace the template. Just make it look like your data comes from the git-revision plugin.

```j2
{% set created = page.meta.git_creation_date_localized %}
```
> This property doesn't only contain the raw date, but the whole [`<span>`](https://github.com/timvink/mkdocs-git-revision-date-localized-plugin/blob/c511b3f76b3fb293e73a0765326888a01226c66b/src/mkdocs_git_revision_date_localized_plugin/util.py#L236) element

As for providing custom dates from your filesystem, you could use a [hook](https://www.mkdocs.org/user-guide/configuration/#hooks)
There are probably different approaches how to read the dates using Python, so please look it up or ask GPT.

Also the git-revisioned plugin [adds](https://github.com/timvink/mkdocs-git-revision-date-localized-plugin/blob/c511b3f76b3fb293e73a0765326888a01226c66b/src/mkdocs_git_revision_date_localized_plugin/plugin.py#L140-L143) the timeago.js library to let it handle the time conversion on the page. There is also a custom timeago fix to make it more compatible with the material theme.

Keep in mind that the filesystem creation and modification date is typically only valid for the same source system and file system type, hence if you decide to backup your files and later restore them from the backup, then some date information could be lost in the process.

**Reply by Atlas-Night-Out** [reply]
> The [`git-revision-date-localized`](https://squidfunk.github.io/mkdocs-material/setup/adding-a-git-repository/#revision-date-localized) plugin might be what you're looking for.

I have tried for a week now to try to get this to work within a mkdocs Material theme and nothing at all shows up. any further helo be great or another solution as I have tried to understand the documents on this too, unless I'm missing somthing, but I post it up to github to  gh-pages page, I also followed or try to follow at this link 
[https://timvink.github.io/mkdocs-git-revision-date-localized-plugin/](mkdocs-git-revision-date-localized-plugin)

**Reply by kamilkrzyskow** [reply]
Since you've created a sibling discussion let's continue there:
- https://github.com/squidfunk/mkdocs-material/discussions/8311
